axum/src/extract/matched_path.rs

```diff
@@ -81,6 +82,21 @@ where
     }
 }
 
+#[async_trait]
+impl<S> OptionalFromRequestParts<S> for MatchedPath
+where
+    S: Send + Sync,
+{
+    type Rejection = MatchedPathRejection;
```

Comment:
I think this should be `Infallible` as missing is the only failure case.

And just to be sure, it seems to me like the only way `MatchedPath` would be missing is if it is accessed inside a fallback handler? If that's so, I think it would be better not to add this impl and have users just use `Result` if they want to guard against using it there.